src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -173,13 +172,22 @@ public Task ExecuteChildContentAsync()
         /// Child content is only executed once. Successive calls to this method or successive executions of the 
         /// returned <see cref="Task{string}"/> return a cached result.
         /// </remarks>
-        public async Task<string> GetChildContentAsync()
+        public async Task<TagHelperContent> GetChildContentAsync()
         {
             if (_childContent == null)
             {
                 _startWritingScope();
                 await _executeChildContentAsync();
-                _childContent = _endWritingScope().ToString();
+                var tagHelperContent = _endWritingScope() as TagHelperContentWrapperTextWriter;
```

Comment:
Since I changed the way of copying from `StringCollectionTextWriter` to `TagHelperContent`, I decided to leave this as such.